test/node_watcher_qc.erl

```diff
@@ -44,47 +44,61 @@ qc_test_() ->
     {timeout, 120, fun() -> ?assert(eqc:quickcheck(?QC_OUT(prop_main()))) end}.
 
 prop_main() ->
+    ?SETUP(
+        fun setup_cleanup/0,
+        ?FORALL(Cmds, commands(?MODULE),
+                begin
+                    %% Setup ETS table to recv broadcasts
+                    ets:new(?MODULE, [ordered_set, named_table, public]),
+                    ets:insert_new(?MODULE, {bcast_id, 0}),
+
+                    %% Start the watcher
+                    {ok, Pid} = riak_core_node_watcher:start_link(),
+
+                    %% Internal call to the node watcher to override default broadcast mechanism
+                    gen_server:call(riak_core_node_watcher, {set_bcast_mod, ?MODULE, on_broadcast}),
```

Comment:
I would prefer if we added an exported function, `riak_core_node_watcher:set_broadcast_module(Module)` and wrap it with a `-ifdef(TEST)`, so that it is only available for testing. 

This will make it possible for us to change the implementation of the `riak_core_node_watcher` without having to change the tests as well.

We might as well clean that up while we are at it.